R/input-submit.R

```diff
@@ -0,0 +1,161 @@
+#' @param placeholder The placeholder text to display when the input is empty.
+#'   This can be used to provide a hint or example of the expected input.
+#' @param value The initial input text. Note that, unlike [textAreaInput()],
+#'   this won't set a server-side value until the value is explicitly submitted.
+#' @param ... Currently not used.
+#' @param button A [tags] element to use for the submit button. It's recommended
+#'   that this be a [input_task_button()] since it will automatically provide a
+#'   busy indicator (and disable) until the next flush occurs. Note also that if
+#'   the submit button launches a [ExtendedTask], this button can also be bound
+#'   to the task ([bind_task_button()]) and/or manually updated for more
+#'   accurate progress reporting ([update_task_button()]).
+#' @param width Any valid CSS unit (e.g., `width="100%"`).
+#' @param submit_key A character string indicating what keyboard event should
+#'   trigger the submit button. The default is `enter`, which will submit the
+#'   input when the user presses the Enter/Return key. The `enter+modifier`
+#'   option will submit the input when the user presses the Enter key while
+#'   holding down Ctrl/Cmd.
+#'
+#' @return A textarea input control that can be added to a UI definition.
+#'
+#' @seealso [update_submit_textarea()], [input_task_button()]
+#'
+#' @examplesIf rlang::is_interactive()
+#'
+#' ui <- page_fluid(
+#'   input_submit_textarea("text", placeholder = "Enter some input..."),
+#'   verbatimTextOutput("value")
+#' )
+#' server <- function(input, output) {
+#'   output$value <- renderText({
+#'     req(input$text)
+#'     Sys.sleep(2)
+#'     paste("You entered:", input$text)
+#'   })
+#' }
+#' shinyApp(ui, server)
+#'
+#' @section Server value:
+#' A character string of the text input. The default value is `""` even if
+#' `value` is provided. The value will only be set/updated when the user submits
+#' the input by pressing the Enter key or clicking the submit button.
+#'
+#' @export
+input_submit_textarea <- function(
+  id,
+  label = NULL,
+  placeholder = NULL,
+  value = "",
+  ...,
+  button = NULL,
+  width = "min(680px, 100%)",
+  submit_key = c("enter", "enter+modifier")
+) {
+  rlang::check_installed("shiny", version = "1.11.1")
+  rlang::check_dots_empty()
+
+  value <- shiny::restoreInput(id = id, default = value)
+  if (length(value) != 1 || !is.character(value)) {
+    stop("`value` must be a character string", call. = FALSE)
+  }
+
+  submit_key <- rlang::arg_match(submit_key)
+  needs_modifier <- isTRUE(submit_key == "enter+modifier")
+
+  if (is.null(button)) {
+    button <- input_task_button(
+      id = paste0(id, "_submit"),
+      class = "btn-sm",
+      label = "Submit \U23CE",
+      title = "Press Enter to Submit",
+      `aria-label` = "Press Enter to Submit",
+      # The label/title above are updated in JS if a modifier key is needed
+      # (since it's platform dependent)
+      `data-default-button` = ""
```

Comment:
Is this attribute still needed or is it a vestige of the submit button?

Reply:
It's needed as a flag for the client-side logic to add a platform-specific modifier key symbol